curation-migration-datasync/index.ts

```diff
@@ -1,22 +1,57 @@
 import config from './config';
 import * as Sentry from '@sentry/serverless';
 import { writeClient } from './database/dbClient';
-import { EventBridgeEvent } from 'aws-lambda';
-import { addScheduledItem } from './eventConsumer';
-import { EventDetailType } from './types';
+import {
+  SQSEvent,
+  SQSBatchResponse,
+  SQSBatchItemFailure,
+  EventBridgeEvent,
+} from 'aws-lambda';
+import { addScheduledItem, removeScheduledItem } from './eventConsumer';
+import { EventDetailType, ScheduledItemPayload } from './types';
 
-export async function handlerFn(event: EventBridgeEvent<any, any>) {
-  console.log(JSON.stringify(event));
+/**
+ * Handler entrypoint. Loops over every record in the message and calls
+ * the function logic to handle the record. Reports the batch failures
+ * (partial success is possible).
+ * @param event the SQS message
+ * @returns the batch failures, if any
+ */
+export async function handlerFn(event: SQSEvent): Promise<SQSBatchResponse> {
+  const batchFailures: SQSBatchItemFailure[] = [];
+  for await (const record of event.Records) {
+    const message = JSON.parse(record.body);
+    console.log(message); // TODO: Remove this later
+    try {
+      await _handlerFn(message);
+    } catch (error) {
+      batchFailures.push({ itemIdentifier: record.messageId });
+      console.log(error);
+      Sentry.addBreadcrumb({
+        message: `Unable to process message: ${record.body}`,
+      });
+      Sentry.captureException(error);
+    }
+  }
+  return { batchItemFailures: batchFailures };
```

Comment:
I think it depends on how Lambda is configured. By default the SQS message contains an array of Records, but it might be possible to configure it to be a singleton array.

Reply:
ya that works for our use case.